goss-testing/scripts/check_bgp_neighbors_established.sh

```diff
@@ -49,19 +49,46 @@ function err_exit
+
+# check if metalLB configmap exists
+kubectl -n metallb-system get cm metallb 2>&1 |grep -i error|wc -l > $TMPFILE ||
+    err_exit 10 "ERROR: Command failed (rc=$?): kubectl -n metallb-system get cm metallb 2>&1 |grep -i error|wc -l " >> $TMPFILE
+metallb_cm=$(kubectl -n metallb-system get cm metallb 2>&1 |grep -i error|wc -l)
+
+# Check for BiCAN configs in MetalLB configmap
+if [ "$metallb_cm" -gt "0" ];then
+	metallb_check="0"
+else
+    kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' | grep customer-management | wc -l >> $TMPFILE ||
+        err_exit 20 "ERROR: Command failed (rc=$?): kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' | grep customer-management | wc -l" > TMPFILE
+	metallb_check=$(kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' | grep customer-management | wc -l)
+fi
+
+# check SLS Networks data for BiCAN toggle
+curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l >> $TMPFILE ||
+    err_exit 30 "ERROR: Command failed (rc=$?): curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l" >> $TMPFILE
+sls_network_check=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN|jq -r .ExtraProperties.SystemDefaultRoute | grep -e CHN -e CAN |wc -l)
+
+# check spine switch motd for csm version
+ssh -o ConnectTimeout=1 -o BatchMode=yes-o StrictHostKeyChecking=no -o PasswordAuthentication=no admin@sw-spine-001 "" 2>&1 | grep CSM | awk '{ print $4 }' >> $TMPFILE ||
+    err_exit 40 "ERROR: Command failed (rc=$?): ssh -o ConnectTimeout=1 -o BatchMode=yes-o StrictHostKeyChecking=no -o PasswordAuthentication=no admin@sw-spine-001 "" 2>&1 | grep CSM | awk '{ print $4 }' " >> $TMPFILE
+switch_motd_check=$(ssh -o ConnectTimeout=1 -o BatchMode=yes -o StrictHostKeyChecking=no -o PasswordAuthentication=no admin@sw-spine-001 "" 2>&1 | grep CSM | awk '{ print $4 }' )
 
 # We really shouldn't be passing a password in plaintext on the command line, but as long as we are, at least
 # we won't also include it in the error message on failure
-if grep -q customer-management $TMPFILE ; then
-    echo "Running: canu validate network bgp --network all --password XXXXXXXX"
-    canu validate network bgp --network all --password {{.Env.SW_ADMIN_PASSWORD}} ||
-        err_exit 20 "canu validate network bgp --network all failed (rc=$?)"
-else
+# RFE: https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-880
+
+if [ "$metallb_check" -eq "0" ] && [ "$sls_network_check" -eq "0" ] || [ "$switch_motd_check" == "1.0" ];then
```

Comment:
On an upgrade, SLS and switch are going to be at 1.2 right away but metallb is going to be at 1.0.
Therefore

( 1 && 0) || 0 == 0 so it will try to do all networks on an upgrade.

I think you want all ORs here.  If any of those checks indicate 1.0, then we need to only do NMN.